components/gitpod-db/src/typeorm/workspace-db-impl.ts

```diff
@@ -462,24 +468,34 @@ export class TypeORMWorkspaceDBImpl extends TransactionalDBImpl<WorkspaceDB> imp
         // The query basically selects all workspace instances for the given owner, whose startDate is within the period, and which are either:
         //  - not stopped yet, or
         //  - is stopped or stopping.
-        const sessions = await workspaceInstanceRepo
+        const sessions = (await workspaceInstanceRepo
             .createQueryBuilder("wsi")
             .leftJoinAndMapOne("wsi.workspace", DBWorkspace, "ws", "ws.id = wsi.workspaceId")
+            .leftJoinAndMapOne("wsi.metrics", DBWorkspaceInstanceMetrics, "wsim", "wsim.instanceId = wsi.id")
             .where("ws.organizationId = :organizationId", { organizationId })
             .andWhere("wsi.creationTime >= :periodStart", { periodStart: periodStart.toISOString() })
             .andWhere("wsi.creationTime <= :periodEnd", { periodEnd: periodEnd.toISOString() })
             .orderBy("wsi.creationTime", "DESC")
             .skip(offset)
             .take(limit)
-            .getMany();
-
-        const resultSessions: { instance: WorkspaceInstance; workspace: Workspace }[] = [];
+            .getMany()) as (DBWorkspaceInstance & {
+            metrics: DBWorkspaceInstanceMetrics | undefined;
+            workspace: DBWorkspace;
```

Comment:
Might be nice to extract this to a `WorkspaceSession` shape, since the type assertion here makes the code look a bit messy.